src/main/java/com/avispl/symphony/dal/infrastructure/management/samsung/smartthings/SamsungSmartThingsAggregatorCommunicator.java

```diff
@@ -1070,13 +1089,18 @@ private void retrieveLocations(boolean retryOnError) {
 
 			LocationWrapper locationWrapper = doGet(request, LocationWrapper.class);
 
-			if (locationWrapper != null && locationWrapper.getLocations() != null) {
+			if (locationWrapper != null) {
 				cachedLocations = locationWrapper.getLocations();
-				locationIdFiltered = findLocationByName(locationFilter).getLocationId();
+				locationIdFiltered = getDefaultValueForNullData(findLocationByName(locationFilter).getLocationId(), SmartThingsConstant.NO_LOCATION_FOUND);
 			} else {
 				logger.error("Locations is empty, Please go to https://graph-ap02-apnortheast2.api.smartthings.com/ to check if the locations are removed from Smartthings");
 			}
+		} catch (FailedLoginException failedLoginException) {
```

Comment:
do you need to rethrow the FailedLoginException? Symphony has an indicator specifically made for that, so maybe it's ok to leave it as FailedLoginException?

Reply:
I fixed it